engine/src/test/java/io/camunda/zeebe/engine/processing/incident/MultiInstanceIncidentTest.java

```diff
@@ -519,6 +519,124 @@ public void shouldCreateIncidentWhenInputCollectionModifiedConcurrently() {
+
+    // then
+    final Record<ProcessInstanceRecordValue> activityEvent =
+        RecordingExporter.processInstanceRecords()
+            .withElementId(ELEMENT_ID)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETING)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Record<IncidentRecordValue> incidentEvent =
+        RecordingExporter.incidentRecords()
+            .withIntent(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    Assertions.assertThat(incidentEvent.getValue())
+        .hasErrorType(ErrorType.EXTRACT_VALUE_ERROR)
+        .hasErrorMessage("failed to evaluate expression 'x': no variable found for name 'x'")
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasElementInstanceKey(activityEvent.getKey())
+        .hasVariableScopeKey(activityEvent.getValue().getFlowScopeKey());
+  }
+
+  @Test
+  public void shouldResolveIncidentDueToCompletionCondition() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess("multi-task")
+                .startEvent()
+                .serviceTask(
+                    ELEMENT_ID,
+                    t ->
+                        t.zeebeJobType(jobType)
+                            .multiInstance(
+                                b ->
+                                    b.parallel()
+                                        .zeebeInputCollectionExpression(INPUT_COLLECTION)
+                                        .zeebeInputElement(INPUT_ELEMENT)
+                                        .completionCondition("=x")))
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId("multi-task")
+            .withVariable(INPUT_COLLECTION, List.of(1, 2, 3))
+            .create();
+
+    completeNthJob(processInstanceKey, 1);
+
+    final List<Record<IncidentRecordValue>> records =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .asList();
+    records.forEach(record -> System.out.println(record.getKey()));
+
+    // an incident is created because the variable `x` is missing
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    // when the missing variable is provided
+    ENGINE.variables().ofScope(processInstanceKey).withDocument(Maps.of(entry("x", true))).update();
+
+    // and we resolve the incident
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+
+    // and complete the other jobs
+    completeNthJob(processInstanceKey, 2);
+    completeNthJob(processInstanceKey, 3);
```

Comment:
❌ We should not have to complete these jobs anymore, because the incident was resolved which continues the processing: retry the complete command on the child instance -> checks the completion condition of the multi-instance body -> terminates the  children (tasks) -> cancels their jobs -> and finally completes the multi-instance body.

Please remove these lines.